pages/advanced-algorithms/available-algorithms/leiden_community_detection.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: leiden_community_detection
+description: Explore Memgraph's Leiden community detection capabilities and learn how to analyze the structure of complex networks. Access tutorials and comprehensive documentation to enhance your understanding of Leiden community detection algorithm.
+---
+
+import { Steps } from 'nextra/components'
+import { Callout } from 'nextra/components'
+import { Card, Cards } from 'nextra/components'
+import GitHub from '/components/icons/GitHub'
+
+# leiden_community_detection
+
+Community in graphs mirrors real-world communities, like social circles. In a
+graph, communities are sets of nodes. M. Girvan and M. E. J. Newman note that
+nodes in a community connect more intensely with each other than with outside
+nodes.
+
+This module employs the [Leiden
+algorithm](https://en.wikipedia.org/wiki/Leiden_algorithm) for community detection
+based on paper [*From Louvain to Leiden: guaranteeing well-connected communities*](https://arxiv.org/abs/1810.08473). 
+The Leiden algorithm is a hierarchical clustering algorithm, that recursively merges communities into single nodes by greedily optimizing the modularity and the process repeats in the condensed graph.
+It enhances the Louvain algorithm by addressing its limitations, particularly in situations where some identified communities are not well-connected. 
+This improvement is made by periodically subdividing communities into smaller, well-connected groups.
+With an $\mathcal{O}(Lm)$ runtime for $m$ edges and $L$ number of iterations, it suits large graphs.
+
+<Cards>
+  <Card
+    icon={<GitHub />}
+    title="Source code"
+    href="https://github.com/memgraph/mage/blob/main/cpp/leiden_community_detection_module/leiden_community_detection_module.cpp"
+  />
+</Cards>
+
+| Trait                    | Value                                                                                                       |
+| ------------------------ | --------------------- |
+| **Module type**          | algorithm             |
+| **Implementation**       | C++                   |
+| **Graph direction**      | undirected            |
+| **Relationship weights** | weighted / unweighted |
+| **Parallelism**          | parallel              |
+
+## Procedures
+
+<Callout type="info">
+You can execute this algorithm on [graph projections, subgraphs or portions of the graph](/advanced-algorithms/run-algorithms#run-procedures-on-subgraph).
+</Callout>
+
+### `get()`
+
+Computes graph communities using the Leiden algorithm.
+
+{<h4> Input: </h4>}
+
+- `subgraph: Graph` (**OPTIONAL**) ➡ A specific subgraph, which is an [object of type Graph](/advanced-algorithms/run-algorithms#run-procedures-on-subgraph) returned by the `project()` function, on which the algorithm is run.
+- `weight: string (default=null)` ➡ Specifies the default relationship weight. If not set, 
```

Comment:
Why is this a string and not a float? Does it refer to a property name?

Reply:
Yes, that is right. We defined this attribute in the same way for other algorithms.

Reply:
That isn't clear. Better would be "Specifies the name of the property which contains the default relationship weight".